tabbycat/api/serializers.py

```diff
@@ -64,3 +118,51 @@ def update(self, instance, validated_data):
         else:
             self.instance.speaker_set.set(speakers)
         return self.instance
+
+
+class TeamSpeakerSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Speaker
+        fields = ('id','name', 'gender','email','phone','anonymous','pronoun',
+                  'categories')
+
+
+class AdjudicatorSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = Adjudicator
+        fields = ('id','name', 'gender','email','phone','anonymous','pronoun',
+                  'institution','base_score')
+
+
+class TeamSerializer(serializers.ModelSerializer):
+    # tournament = serializers.SlugRelatedField(
+    #     queryset=Tournament.objects.all(),
+    #     slug_field='slug'
+    # )
+    speakers = TeamSpeakerSerializer(many=True)
+
+    class Meta:
+        model = Team
+        fields = ('id','reference', 'code_name',
+                  'institution', 'speakers','use_institution_prefix','break_categories')
+
+    def create(self,validated_data):
+        speaker_data = validated_data.pop('speakers')
+        team = Team.objects.create(**validated_data)
+        for i in speaker_data:
+            Speaker.objects.create(team=team, **i)
+        return team
+
+
+class InstitutionSerializer(serializers.ModelSerializer):
+
+    teams = serializers.SerializerMethodField()
```

Comment:
According to Django reference, `queryset` is "used for model instance lookups when validating the field input. " It does not serve as a filter for representations. I did notice your implementation for SpeakerEligibilitySerializer, though.

Reply:
(Django REST reference) I see. Now I have stuff to fix! :) Good catch; I was oblivious to this problem, but I found a more elegant solution.

I am still opposed to using `MethodField` as it is very low-level, thus requires us to create our own logic. It may also prevent us from using the field for writing, if that becomes a concern. Using a method field is inelegant...

To avoid needing it, in `InstitutionViewSet`, you can use the `.prefetch_related()` method of querysets with a [`Prefetch()`](https://docs.djangoproject.com/en/3.0/ref/models/querysets/#prefetch-objects) object to filter `team_set` to only the relevant teams (like `self.tournament.team_set.all()`). It is also better for the DB as it will require fewer queries.

Reply:
I decided to mark this conversation as unresolved because Prefetch does not work properly for detail views. In detail views for institutions, all teams get returned while in list views, they get filtered beforehand. This probably has something to do with the mechanism of Prefetch. We need either overwrite `retrieve()` method or `get_object()` method in the `InstitutionViewSet `, or reemploy `MethodField`.

Reply:
I haven't tested this with Django REST, but this problem would be very odd as we have the expected result when doing `.get()` on a query set with a prefetch (that I tested), and DRF does just that by using `.get()` on the queryset.

Reply:
There was a mismatch between my test data and my test database, this was a false alarm